thrall/app/lib/ElasticSearch.scala

```diff
@@ -218,21 +220,39 @@ class ElasticSearch(config: ThrallConfig, metrics: ThrallMetrics) extends Elasti
       .incrementOnFailure(metrics.failedCollectionsUpdates) { case e: VersionConflictEngineException => true }
     }
 
-  def prepareImageUpdate(id: String)(op: UpdateRequestBuilder => Future[UpdateResponse]): List[Future[UpdateResponse]] = {
-    prepareForMultipleIndexes( index => {
-          val updateRequest = client.prepareUpdate(index, imageType, id)
-            .setScriptLang("groovy")
-          op(updateRequest)
-        }
-    )
+  def getImageAlbum(id: String)(implicit ex: ExecutionContext): Future[Option[String]] =
+    client.prepareGet(readAlias, imageType, id)
+      .setFetchSource("userMetadata.album.title", null)
+      .executeAndLog(s"get album for image $id")
+      .map{res =>
+        res.sourceOpt.map(json => (json \ "userMetadata" \ "album" \ "title").as[String])}
+
+  def getImageIdsInAlbum(albumNameOpt: Option[String])(implicit ex: ExecutionContext): Future[List[String]] = albumNameOpt match {
+    case Some(albumName) =>
+      val q = filteredQuery(
+        matchQuery("userMetadata.album.title", albumName),
+        andFilter(
+          missingOrEmptyFilter("syndicationRights"))
```

Comment:
Shouldn't we overwrite the syndication rights info everywhere based on a RCS update?

Reply:
That's what I initially thought. But I think we need a bit more logic in that case as at the moment I'd be overriding everything including the `acquired` field. I don't think we should change a `no` to a `yes`. I'll have a think how to only apply the rights properties.